src/debugger/VariableFactory.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { VariableContainer, Variable } from "./types.js";
+import { ReferenceMap } from "./ReferenceMap.js";
+import { LexicalEnvironment, DeclarativeEnvironmentRecord } from "./../environment.js";
+import { Value, ConcreteValue, PrimitiveValue } from "./../values/index.js";
+
+export class VariableFactory {
```

Comment:
1. Add comments for several concepts here: the purpose of this class and how should client use it. And the meaning of "reference".
2. I would call this "VariableManager", Factory normally means the factory pattern which creates variable well this class does not create variable only manages its lifetime.
3. You need to clear the content of the map during "resume" otherwise, you are leaking memory.